### PR TITLE
Project filters2

### PR DESCRIPTION
**Pull request by oorion**
NOTE: Must bundle and rake db:migrate

### CODE REVIEW

File: app/controllers/tenants/projects_controller.rb [hunk 1]
```diff
@@ -58,6 +58,7 @@ def project_params
                                     :requested_by,
                                     :retired, :tenant_id, :repayment_begin,
                                     :repayment_rate, categories: [])
+                                    # photos: {image: [:image_file_name, :image_content_type]}
```
Comment (EmilyMB) [inline app/controllers/tenants/projects_controller.rb]: Can we just delete this line?

File: app/controllers/projects_controller.rb [hunk 2]
```diff
@@ -4,12 +4,17 @@ def index
       project.tenant.visible_to_lenders == true &&
       project.current_amount_needed > 0
     end
-    @categories = Category.all
+    @categories = Category.select(:name).all
+
+    if request.xhr?
+      require 'pry' ; binding.pry
```
Comment (EmilyMB) [inline app/controllers/projects_controller.rb]: I think this pry needs to go!

**Comment by EmilyMB** [comment]
This was a mammoth amount of work! I know we need to get this merged. My suggestions:
1. Remove the binding.pry
2. Remove the lend button completely from the projects_path (currently clicking lend on any project will add the first project to the cart). If we can get this patched later, great, but I think we should get the bulk of the changes merged in asap.
3. Change the default for projects_path to have all categories selected for the filter. If this is complicated, have the Keevahh link in the navbar link to root, rather than projects_path

File: Gemfile [hunk 3]
```diff
@@ -43,6 +43,7 @@ gem "less-rails",               '~> 2.6.0'
 gem "rails_12factor",           '~> 0.0.3'
 gem 'paperclip',                '~> 4.2.1'
 gem 'faker',                    '~> 1.4.2'
+gem 'aws-sdk'
```
Comment (mikedao) [inline Gemfile]: I assume this is the supporting gem for S3?

File: app/controllers/projects_controller.rb (hunk 2 above)
Comment (mikedao) [inline app/controllers/projects_controller.rb]: I concur. We probably don't need this. Probably.

File: app/controllers/sessions_controller.rb [hunk 4]
```diff
@@ -38,7 +38,7 @@ def authenticate_user(user)
   end
 
   def invalid_login
-    flash[:errors] = "Invalid Login"
+    flash[:error] = "Invalid Login"
```
Comment (mikedao) [inline app/controllers/sessions_controller.rb]: Why this change?

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -2,7 +2,7 @@ class UsersController < ApplicationController
   def new
     store_location
     if params[:message]
-      flash[:notice] = params[:message]
+      flash[:alert] = params[:message]
```
Comment (mikedao) [inline app/controllers/users_controller.rb]: Are these changes for style purposes?

**Comment by mikedao** [comment]
There were added unskipped tests - are these okay?

**Comment by oorion** [comment]
Removed pry.  Also update Lend button to work for all projects on project index page.

**Comment by oorion** [comment]
I agree that when no filters are selected that it should show all projects, but I think that should be on another branch.

**Comment by oorion** [comment]
We needed to skip those tests since the page they test relies on javascript and nobody, including the instructors and mentors, could get selenium or webkit working with minitest.

File: Gemfile (hunk 3 above)
Comment (oorion) [inline Gemfile]: Yes, it is.

File: app/controllers/users_controller.rb (hunk 5 above)
Comment (oorion) [inline app/controllers/users_controller.rb]: Yes.

**Comment by mikedao** [comment]
Merge!
